unit-tests/dds/test-stream-sensor-bridge.py

```diff
@@ -0,0 +1,341 @@
+        RuntimeError, "profile <'Infrared 1' 1280x800 GREY @ 30 Hz> is incompatible with already-open <'Depth' 1280x720 Z16 @ 30 Hz>" )
+    bridge.open( find_server_profile( 'Infrared 1', '1280x720 GREY @ 30 Hz' ))  # same profile, makes it explicit!
+    bridge.commit()
+    test.check_equal( len(active_sensors), 0 )  # not streaming yet
+    start_stream( 'Depth' )
+    ( test.check_equal( len(active_sensors), 1 )
+        and test.check_equal( next(iter(active_sensors)), 'Stereo Module' )
+        and test.check_equal( len(active_sensors['Stereo Module']), 3 )
+        )
+    bridge.open( find_active_profile( 'Infrared 1' ))  # already explicit, same profile: does nothing
+    start_stream( 'Infrared 2' )  # starts it implicitly
+except Exception:
+    test.unexpected_exception()
+finally:
+    reset()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "stream profiles reset" )
+try:
+    bridge.open( find_server_profile( 'Infrared 1', '640x480 GREY @ 60 Hz' ))
+    bridge.add_implicit_profiles()                     # adds Depth, IR2
+    bridge.commit()
+    test.check_equal( len(active_sensors), 0 )  # not streaming yet
+    start_stream( 'Infrared 1' )
+    ( test.check_equal( len(active_sensors), 1 )
+        and test.check_equal( next(iter(active_sensors)), 'Stereo Module' )
+        and test.check_equal( len(active_sensors['Stereo Module']), 3 )
+        )
+    test.check_equal( find_active_profile( 'Infrared 2' ).to_string(), "<'Infrared 2' 640x480 GREY @ 60 Hz>" )
+    stop_stream( 'Infrared 1' )
+    test.check_equal( len(active_sensors), 0 )  # not streaming again
+    # We don't reset - last commit should still stand!
+    start_stream( 'Infrared 2' )
+    ( test.check_equal( len(active_sensors), 1 )
+        and test.check_equal( next(iter(active_sensors)), 'Stereo Module' )
+        and test.check_equal( len(active_sensors['Stereo Module']), 3 )
+        )
+    test.check_equal( find_active_profile( 'Infrared 2' ).to_string(), "<'Infrared 2' 640x480 GREY @ 60 Hz>" )
+    stop_stream( 'Infrared 2' )
+    test.check_equal( len(active_sensors), 0 )  # not streaming again
+    # Now reset - commit should be lost and we should be back to the default profile
+    bridge.reset()
+    start_stream( 'Infrared 2' )
+    test.check_equal( find_active_profile( 'Infrared 2' ).to_string(), servers['Infrared 2'].default_profile().to_string() )
+except Exception:
+    test.unexpected_exception()
+finally:
+    reset()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "two different sensors" )
+try:
+    bridge.open( servers['Depth'].default_profile() )  # 1280x720 Z16 @ 30 Hz
+    bridge.add_implicit_profiles()                     # adds IR1, IR2
+    bridge.commit()
+    start_stream( 'Depth' )
+    # We have a stream streaming; reset shouldn't touch it
+    bridge.reset()
```

Comment:
Shouldn't reset return some failure?

Reply:
reset() works on streams that aren't streaming; it only gives a warning for streams that are.
Otherwise, think about it:
* start Depth (from the Viewer)
* start Color
The second `open-streams` will try to reset() and it'll throw.

Reply:
OK, I see now the LOG_WARNING. Haven't looked at the implementation yet, started with header and test. Thought there could be a boolean return value to signal if it did something or not. Warning is also accepptible